test/jdk/java/awt/Window/LocationByPlatform/TestLocationByPlatform.java

```diff
@@ -60,58 +61,45 @@ offset. Another has its location explicitly set to (0, 450).
             .rows(13)
             .columns(40)
             .build();
-        EventQueue.invokeAndWait(() -> createUI());
+        EventQueue.invokeAndWait(TestLocationByPlatform::createUI);
         passFailJFrame.awaitAndCheck();
     }
     private static void createUI() {
         Frame frame = new Frame("Normal");
         frame.setLocation(0, 450);
-        Canvas c = new Canvas() {
-            public Dimension getPreferredSize() {
-                return new Dimension(400, 400);
-            }
-
-            public void paint(Graphics g) {
-                g.setColor(Color.red);
-                for (int i = 399; i >= 0; i--) {
-                    g.drawLine(400 - i - 1, 400 - i - 1,
-                        400 - i - 1, 399);
-                }
-                g.setColor(Color.blue);
-                g.drawLine(0, 0, 399, 0);
-                g.drawLine(0, 0, 0, 399);
-                g.drawLine(0, 399, 399, 399);
-                g.drawLine(399, 0, 399, 399);
-            }
-        };
+        Canvas c = new MyCanvas();
         frame.add(c, BorderLayout.CENTER);
         frame.pack();
         PassFailJFrame.addTestWindow(frame);
         frame.setVisible(true);
 
         frame = new Frame("Location by platform");
         frame.setLocationByPlatform(true);
-        c = new Canvas() {
-            public Dimension getPreferredSize() {
-                return new Dimension(400, 400);
-            }
-
-            public void paint(Graphics g) {
-                g.setColor(Color.red);
-                for (int i = 399; i >= 0; i--) {
-                    g.drawLine(400 - i - 1, 400 - i - 1,
-                        400 - i - 1, 399);
-                }
-                g.setColor(Color.blue);
-                g.drawLine(0, 0, 399, 0);
-                g.drawLine(0, 0, 0, 399);
-                g.drawLine(0, 399, 399, 399);
-                g.drawLine(399, 0, 399, 399);
-            }
-        };
+        c = new MyCanvas();
         frame.add(c, BorderLayout.CENTER);
         frame.pack();
         PassFailJFrame.addTestWindow(frame);
         frame.setVisible(true);
     }
 }
+
+class MyCanvas extends Canvas {
```

Comment:
Please make a nested static class inside `TestLocationByPlatform`.

It avoids any potential conflicts for any other tests which may use `MyCanvas` that you may have in classpath in your IDE.

Reply:
I understand, updated the code.